packages/core/src/clients/decorators/public.ts

```diff
@@ -0,0 +1,62 @@
+import { CosmWasmClient } from '@cosmjs/cosmwasm-stargate'
+import { getAccountFactoryAddress } from '../../actions/public/account-factory/get-account-factory-address.js'
+import { getAccountFactoryQueryClient } from '../../actions/public/account-factory/get-account-factory-query-client'
+import { getAccountFactoryQueryClientFromApi } from '../../actions/public/account-factory/get-account-factory-query-client-from-api'
+import { getAnsHostAddress } from '../../actions/public/ans/get-ans-host-address.js'
+import { getAnsHostQueryClientFromApi } from '../../actions/public/ans/get-ans-host-query-client-from-api.js'
+import { getAnsHostQueryClient } from '../../actions/public/ans/get-ans-host-query-client.js'
+import { getVersionControlModuleData } from '../../actions/public/version-control/get-version-control-module-data.js'
+import { getVersionControlQueryClientFromApi } from '../../actions/public/version-control/get-version-control-query-client-from-api.js'
+import { SliceFirst, SliceFirstTwo } from '../../types/utils.js'
+import { Client } from '../create-client.js'
+
+export type PublicActions = {
+  getAccountFactoryAddress(
+    ...args: SliceFirst<Parameters<typeof getAccountFactoryAddress>>
+  ): ReturnType<typeof getAccountFactoryAddress>
+  getAccountFactoryQueryClient(
+    ...args: SliceFirst<Parameters<typeof getAccountFactoryQueryClient>>
+  ): ReturnType<typeof getAccountFactoryQueryClient>
+  getAccountFactoryQueryClientFromApi(
+    ...args: SliceFirstTwo<
+      Parameters<typeof getAccountFactoryQueryClientFromApi>
+    >
+  ): ReturnType<typeof getAccountFactoryQueryClientFromApi>
```

Comment:
To confirm - we're exposing both the "raw" and "api-based" client retrievals so that we can use this the former in building the Subgraph or just to allow more granular access to devs?